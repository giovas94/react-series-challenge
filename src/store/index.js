import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import { persistStore } from "redux-persist";

import reducers from "../reducers";
import rootSaga from "../sagas";
import customMiddleware from "../middleware";

// create the saga middleware
const sagaMiddleware = createSagaMiddleware();

// dev tools middleware
const reduxDevTools =
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();

const initialState = {};
const middleware = [sagaMiddleware, customMiddleware];

// create a redux store with our reducer above and middleware
const store = createStore(
  reducers,
  initialState,
  compose(
    applyMiddleware(...middleware),
    reduxDevTools
  )
);

const persistor = persistStore(store);

// run the saga
sagaMiddleware.run(rootSaga);

export { store, persistor };
