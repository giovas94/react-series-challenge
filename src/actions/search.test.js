import { store } from "../store";
import searchActions from "./search";

describe("search_actions", () => {
  describe("search_input", () => {
    test("Should dispatches the correct action and payload", () => {
      store.dispatch(searchActions.creators.setSearchInput("soccer"));
      expect(store.getState()).toMatchSnapshot();
    });
  });

  describe("search_favorite", () => {
    test("Should dispatches the correct action and payload", () => {
      store.dispatch(searchActions.creators.setSearchFavorite("google"));
      expect(store.getState()).toMatchSnapshot();
    });
  });
});
