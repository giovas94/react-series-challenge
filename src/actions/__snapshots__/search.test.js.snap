// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search_actions search_favorite Should dispatches the correct action and payload 1`] = `
Object {
  "_persist": Object {
    "rehydrated": true,
    "version": -1,
  },
  "favorites": Object {
    "_persist": Object {
      "rehydrated": true,
      "version": -1,
    },
    "filtered": null,
    "items": Object {},
  },
  "search": Object {
    "searchFavorite": "",
    "searchInput": "google",
  },
  "searchGifs": Object {
    "data": null,
    "error": null,
    "fetching": false,
  },
  "trendGifs": Object {
    "data": null,
    "error": null,
    "fetching": false,
  },
}
`;

exports[`search_actions search_input Should dispatches the correct action and payload 1`] = `
Object {
  "_persist": Object {
    "rehydrated": true,
    "version": -1,
  },
  "favorites": Object {
    "_persist": Object {
      "rehydrated": true,
      "version": -1,
    },
    "filtered": null,
    "items": Object {},
  },
  "search": Object {
    "searchFavorite": "",
    "searchInput": "soccer",
  },
  "searchGifs": Object {
    "data": null,
    "error": null,
    "fetching": false,
  },
  "trendGifs": Object {
    "data": null,
    "error": null,
    "fetching": false,
  },
}
`;
